src/include/duckdb/common/arrow/appender/varchar_data.hpp

```diff
@@ -108,7 +109,77 @@ struct ArrowVarcharData {
 	static void Finalize(ArrowAppendData &append_data, const LogicalType &type, ArrowArray *result) {
 		result->n_buffers = 3;
 		result->buffers[1] = append_data.main_buffer.data();
-		result->buffers[2] = append_data.aux_buffer.data();
+		result->buffers[2] = append_data.aux_buffer[0].data();
+	}
+};
+
+struct ArrowVarcharToStringViewData {
+	static void Initialize(ArrowAppendData &result, const LogicalType &type, idx_t capacity) {
+		result.main_buffer.reserve((capacity) * sizeof(arrow_string_view_t));
+		result.aux_buffer[0].reserve(capacity);
+		result.aux_buffer[1].reserve(sizeof(int64_t));
+	}
+
+	static void Append(ArrowAppendData &append_data, Vector &input, idx_t from, idx_t to, idx_t input_size) {
+		idx_t size = to - from;
+		UnifiedVectorFormat format;
+		input.ToUnifiedFormat(input_size, format);
+
+		// resize the validity mask and set up the validity buffer for iteration
+		ResizeValidity(append_data.validity, append_data.row_count + size);
+		auto validity_data = (uint8_t *)append_data.validity.data();
+
+		append_data.main_buffer.resize(append_data.main_buffer.size() + sizeof(arrow_string_view_t) * (size));
+		// resize the offset buffer - the offset buffer holds the offsets into the child array
+		auto data = UnifiedVectorFormat::GetData<string_t>(format);
+		for (idx_t i = from; i < to; i++) {
+			auto result_idx = append_data.row_count + i - from;
+			auto arrow_data = append_data.main_buffer.GetData<arrow_string_view_t>();
+			auto source_idx = format.sel->get_index(i);
+			if (!format.validity.RowIsValid(source_idx)) {
+				// Null value
+				uint8_t current_bit;
+				idx_t current_byte;
+				GetBitPosition(result_idx, current_byte, current_bit);
+				SetNull(append_data, validity_data, current_byte, current_bit);
+				continue;
+			}
+			// These two are now the same buffer
+			idx_t string_length = ArrowVarcharConverter::GetLength(data[source_idx]);
+			auto string_data = data[source_idx].GetData();
+			if (string_length <= arrow_string_view_t::max_inlined_bytes) {
+				//	This string is inlined
+				//  | Bytes 0-3  | Bytes 4-15                            |
+				//  |------------|---------------------------------------|
+				//  | length     | data (padded with 0)                  |
+				arrow_data[result_idx] = arrow_string_view_t(string_length, string_data);
+			} else {
+				// This string is not inlined, we have to check a different buffer and offsets
+				//  | Bytes 0-3  | Bytes 4-7  | Bytes 8-11 | Bytes 12-15 |
+				//  |------------|------------|------------|-------------|
+				//  | length     | prefix     | buf. index | offset      |
+				arrow_data[result_idx] = arrow_string_view_t(string_length, string_data, 0, append_data.offset);
+				auto current_offset = append_data.offset + string_length;
```

Comment:
Wasn't it said that non-inlined strings should always take up 12 bytes?
This is going off of our string_t size, which could be less than 12, in which case we still need to reserve 12 bytes